File: doc-examples/example-java/src/test/java/example/PricingExpressions.java
package example;

import io.micronaut.el.annotation.ELEnvironment;
import io.micronaut.el.annotation.ELExpression;
import io.micronaut.el.annotation.ELFunctions;
import io.micronaut.el.annotation.ELVariable;

@ELEnvironment(
    variables = @ELVariable(name = "book", type = Book.class),
    functions = @ELFunctions(PricingService.class) // <1>
)
@ELExpression(value = "${pricing:quote(book, 3)}", expectedType = double.class, name = "QUOTE") // <2>
@ELExpression(value = "${pricing:quote(book, 3) += ' ' += pricing:currency()}", expectedType = String.class, name = "PRICED") // <3>
public class PricingExpressions {
}

File: doc-examples/example-python/src/test/python/example/PricingExpressions.py
from micronaut.el.annotation import ELEnvironment, ELExpression, ELVariable

from example.Book import Book


@ELEnvironment(variables=[ELVariable(name="book", type=Book)])  # <1>
@ELExpression(value="${pricing:quote(book, 3)}", expectedType=float, name="QUOTE")  # <2>
@ELExpression(value="${pricing:quote(book, 3) += ' ' += pricing:currency()}", expectedType=str, name="PRICED")  # <3>
class PricingExpressions:
    pass
